### PR TITLE
SOF-6927: handle optional species label in xyz parser

### PR DESCRIPTION
**Pull request by pranabdas**

### CODE REVIEW

File: src/parsers/xyz.ts [hunk 1]
```diff
@@ -60,13 +61,24 @@ export interface ParsedObject {
  */
 function _parseXYZLineAsWords(line: string): ParsedObject {
     const words = s.words(line);
+    const element: string = words[0].replace(/\d$/, ""); // Fe1 => Fe
```
Comment (timurbazhirov) [inline src/parsers/xyz.ts:64]:
```js
const elementWithOptionalLabel = words[0];
const element = elementWithOptionalLabel.replace(...);
```

File: src/parsers/xyz.ts [hunk 2]
```diff
@@ -60,13 +61,24 @@ export interface ParsedObject {
...
-        element: words[0],
+
+    const basisObj: ParsedObject = {
+        element,
         coordinates: [+words[1], +words[2], +words[3]],
         // Below maps zero values to false (atom is fixed) and non-zero values to true (atom is moving)
         constraints: [constraint(+words[4]), constraint(+words[5]), constraint(+words[6])],
     };
+
+    if (element !== words[0]) {
```
Comment (timurbazhirov) [inline src/parsers/xyz.ts:74]: `if (elementWithOptionalLabel !== element)`

File: src/parsers/xyz.ts [hunk 3]
```diff
@@ -60,13 +61,24 @@ export interface ParsedObject {
  */
 function _parseXYZLineAsWords(line: string): ParsedObject {
     const words = s.words(line);
+    const element: string = words[0].replace(/\d$/, ""); // Fe1 => Fe
     const constraint = (n: number) => parseInt(`${n}`, 10) !== 0;
-    return {
-        element: words[0],
+
+    const basisObj: ParsedObject = {
```
Comment (timurbazhirov) [inline src/parsers/xyz.ts:67]: basisLineObject

**Review by timurbazhirov — APPROVED** [review]

File: src/parsers/xyz.ts [hunk 4]
```diff
@@ -117,6 +133,29 @@ function toBasisConfig(txt: string, units = "angstrom", cell = Basis.defaultCell
...
+    });
+
+    if (!_.isEmpty(labels)) {
+        return {
+            ...basisConfig,
+            labels,
+        };
+    }
+
+    return basisConfig;
```
Comment (timurbazhirov) [inline src/parsers/xyz.ts:158]: Let's add more test cases for data with labels

File: tests/parsers/xyz.js [hunk 5]
```diff
@@ -12,3 +12,14 @@ describe("Parsers:XYZ", () => {
         ]);
     });
 });
+
+describe("Parsers:XYZ", () => {
```
Comment (pranabdas) [inline tests/parsers/xyz.js:16]: @timurbazhirov is this sufficient?

File: src/parsers/xyz.ts [hunk 6]
```diff
@@ -134,7 +174,9 @@ function fromBasis(
     const XYZArray: string[] = [];
     basisClsInstance._elements.array.forEach((item, idx) => {
         // assume that _elements and _coordinates are indexed equivalently
-        const element = s.sprintf("%-3s", item);
+        const atomicLabel = basisClsInstance.atomicLabelsArray[idx];
+        const elementWithLabel = item + atomicLabel;
+        const element = s.sprintf("%-3s", elementWithLabel);
```
Comment (timurbazhirov) [inline src/parsers/xyz.ts:179]: As we discussed, let's create a task to generalize the labels, pls, wrt to:
- utilizing any string as label
- making the numeric space larger than just 0-9
- allowing to store metadata (perhaps just the description field, like in webapp) to explain what the labels mean

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**